### PR TITLE
Move SystemAccessControl implementations to presto-plugin-toolkit

### PR DESCRIPTION
**Pull request by kokosing**
Move SystemAccessControl implementations to presto-plugin-toolkit

Thanks to that:
 - they could be used to compose other Presto plugins. This is very
 useful because Presto does not support multiple system access controls,
 so user can merge two implementations into a single one.
 This is nice because this way user manages the semantic how multiple
 system access controls should work.
 - file based access control are placed together
 - implementation of file system access control no longer depends on
 how system access control is used by Presto.
 - tests are simpler

### CODE REVIEW

File: presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java [hunk 1]
```diff
@@ -1,276 +0,0 @@
-/*
```
Comment (kokosing) [inline presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java]: restore this test. It was also testing the `AccessControlManager`

File: presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java [hunk 2]
```diff
@@ -197,21 +196,33 @@ private boolean canAccessCatalog(Identity identity, String catalogName)
     @Override
     public void checkCanCreateSchema(Identity identity, CatalogSchemaName schema)
     {
+        if (!canAccessCatalog(identity, schema.getCatalogName())) {
```
Comment (kokosing) [inline presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java]: It is access control manager contract that ensures that these methods should not be called if user does not have access to catalog.

They don't have to be implemented. Revert.

**Comment by kokosing** [comment]
ping

File: presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/security/TestFileBasedSystemAccessControl.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+import static com.google.common.io.Files.copy;
+import static io.prestosql.plugin.base.security.FileBasedAccessControlConfig.SECURITY_CONFIG_FILE;
+import static io.prestosql.plugin.base.security.FileBasedAccessControlConfig.SECURITY_REFRESH_PERIOD;
+import static io.prestosql.spi.testing.InterfaceTestUtils.assertAllMethodsOverridden;
+import static java.lang.Thread.sleep;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.util.Files.newTemporaryFile;
+
+public class TestFileBasedSystemAccessControl
```
Comment (findepi) [inline presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/security/TestFileBasedSystemAccessControl.java]: In `Move SystemAccessControl implementations to presto-plugin-toolkit` commit -- why this file does not appear as moved?

What with `presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java`?

File: presto-plugin-toolkit/src/test/resources/catalog.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{
+  "catalogs": [
```
Comment (findepi) [inline presto-plugin-toolkit/src/test/resources/catalog.json]: Is this new or moved?

File: presto-plugin-toolkit/src/test/resources/catalog_principal.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{
+  "catalogs": [
```
Comment (findepi) [inline presto-plugin-toolkit/src/test/resources/catalog_principal.json]: Is this new or moved?

File: presto-plugin-toolkit/src/test/resources/security-config-file-with-unknown-rules.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "schemas": [
```
Comment (findepi) [inline presto-plugin-toolkit/src/test/resources/security-config-file-with-unknown-rules.json]: Is this new or moved?

File: presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/security/TestFileBasedSystemAccessControl.java (hunk 3 above)
Comment (kokosing) [reply presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/security/TestFileBasedSystemAccessControl.java]: It was copied and modified a bit. Original tests are testing things on different level (using `AccessControl`), here I am using `FileBasedSystemAccessControl` directly.

File: presto-plugin-toolkit/src/test/resources/catalog.json (hunk 4 above)
Comment (kokosing) [reply presto-plugin-toolkit/src/test/resources/catalog.json]: it is copied

File: presto-plugin-toolkit/src/test/resources/catalog_principal.json (hunk 5 above)
Comment (kokosing) [reply presto-plugin-toolkit/src/test/resources/catalog_principal.json]: it is copied (no changes)

File: presto-plugin-toolkit/src/test/resources/security-config-file-with-unknown-rules.json (hunk 6 above)
Comment (kokosing) [reply presto-plugin-toolkit/src/test/resources/security-config-file-with-unknown-rules.json]: it is copied (no changes)

File: presto-plugin-toolkit/src/test/resources/catalog_principal.json (hunk 5 above)
Comment (kokosing) [reply presto-plugin-toolkit/src/test/resources/catalog_principal.json]: it is copied (no changes)

**Pull request merged**